compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingIntoPackUnPack.cpp

```diff
@@ -608,33 +609,33 @@ struct MaterializeFlowDispatchTensorLoadOp
     }
 
     auto sourceType = loadOp.getSourceType();
-    auto boundTensorType = sourceType.getBoundType();
+    auto boundTensorType = cast<RankedTensorType>(sourceType.getBoundType());
     auto *typeConverter = static_cast<const MaterializeEncodingTypeConverter *>(
         getTypeConverter());
     if (typeConverter->convertType(boundTensorType) == boundTensorType) {
       return rewriter.notifyMatchFailure(loadOp, "bound type already valid");
     }
 
     Location loc = loadOp.getLoc();
+    SmallVector<OpFoldResult> newMixedSizes = getMixedValues(
+        boundTensorType.getShape(), loadOp.getSourceDims(), rewriter);
     FailureOr<SmallVector<OpFoldResult>> convertedMixedSizes =
         getPackedDimsForDispatchTensor(rewriter, loc, *typeConverter,
                                        sourceType, loadOp.getSourceDims(),
                                        this->materializeEncodingValueFn);
-    if (failed(convertedMixedSizes)) {
-      return rewriter.notifyMatchFailure(
-          loadOp, "failed to get converted dynamic dims for result");
+    if (succeeded(convertedMixedSizes)) {
```

Comment:
What happens on failure? I would expect this to return some sort of error?

Reply:
If it fails, it means that the type convert can not deal with encodings properly. I.e., inner_tile_sizes is missing. In this case, all the encodings are dropped. It is safe because it all the operations with such encoding will fail. So it falls back to undo "SetEncoding". I.e., it drops all the encodings.